src/frequenz/channels/_sender.py

```diff
@@ -1,7 +1,53 @@
 # License: MIT
 # Copyright © 2022 Frequenz Energy-as-a-Service GmbH
 
-"""Channel sender and associated exceptions."""
+"""Sender interface and related exceptions.
+
+# Senders
+
+Messages are sent to a [channel](/user-guide/channels) through
+[Sender][frequenz.channels.Sender] objects. [Senders][frequenz.channels.Sender] are
+usually created by calling `channel.new_sender()`, and are a very simple abstraction
+that only provides a single [`send()`][frequenz.channels.Sender.send] method:
+
+```python
+from frequenz.channels import Anycast
+
+channel = Anycast[int](name="test-channel")
+sender = channel.new_sender()
+
+await sender.send("Hello, world!")
+```
+
+Even when [`send()`][frequenz.channels.Sender.send] is an asynchronous method, some
+channels may implement it as a synchronously. For example, buffered channels that
```

Comment:
This is poorly worded, I guess the last "as a" shouldn't be there. Is this better?

> Even when [`send()`][frequenz.channels.Sender.send] is an asynchronous method, some
> channels may implement it synchronously.

Does that makes it clearer?

Reply:
That's probably not much better, I think there is extra confusion because it is not mentioning blocking, which is the main point. What about this?

> Even when [`send()`][frequenz.channels.Sender.send] is an asynchronous method, some
> channels may implement it in a synchronous, non-blocking way.

Reply:
synchronous + non-blocking can be a bit confusing, but if people read further, I think it should become clearer.